src/Microphone.py

```diff
@@ -62,6 +79,25 @@ def stop_recording(self):
         self.stream.close()
         self.wave_file.close()
         self.is_recording = False
+    
+    def idleListen(self):
+        alerted = False
+        RECORD_SECONDS = 5
+        stream = self.pyaudio.open(
+        format=self.format,
+        channels=self.channels,
+        rate=self.framerate,
+        input=True,
+        output=True,
+        frames_per_buffer=self.chunk*2.0
+        )
+        while alerted == False:
+            micData = stream.read(self.chunk)
+            amplitude = getAmplitude(micData)
+            decibels = 20*math.log10(amplitude)
+            if(decibels > 20):
+                publish("Noise Alert", ALERT_ON)
```

Comment:
Usage to publish is as follows: 
publisher = getClient()
publish(publisher, "Noise Alert", Topic.ALERT_ON)

You would probably create the client in the constructor.